torchci/pages/api/drci/drci.ts

```diff
@@ -1110,69 +1110,63 @@ export async function reorganizeWorkflows(
       workflow.name,
     ])
   );
-  const workflowsByPR: Map<number, PRandJobs> = new Map();
-  const headShaTimestamps: Map<string, string> = new Map();
-
-  for (const workflow of dedupedRecentWorkflows) {
-    const prNumber = workflow.pr_number;
-    if (!workflowsByPR.has(prNumber)) {
```

Comment:
Why did this version check for the prNumber and only apply the logic to the entries missing it, while the new version doesn't need it and runs the code against all entries?

Reply:
The PR regroups and reorders things so that the slow fetches can happen outside of the sequential for loop. The same number of api calls should happen, they just happen at a different time.  Here's some pseudo code that kinda explains whats going on


Goal: construct map<pr number, pr info and job list>
Old:
```
for each workflow (sequential):
  add to map
  if workflow's pr hasn't been seen yet (new key), fetch pr info and maybe info about the commit (slow)
```


New:
```
for each workflow (I think sequential):
  add to map
for each key (pr number) in map (parallel):
  fetch pr info and maybe info about commit
```